tests/components/ecobee/test_climate.py

```diff
@@ -428,3 +435,30 @@ async def test_turn_aux_heat_off(hass: HomeAssistant, mock_ecobee: MagicMock) ->
     )
     assert mock_ecobee.set_hvac_mode.call_count == 1
     assert mock_ecobee.set_hvac_mode.call_args == mock.call(0, "auto")
+
+
+async def test_preset_indefinite_away(ecobee_fixture, thermostat) -> None:
+    """Test indefinite away showing correctly, and not as temporary away."""
+    ecobee_fixture["program"]["currentClimateRef"] = "away"
+    ecobee_fixture["events"][0]["holdClimateRef"] = "away"
+    assert thermostat.preset_mode == "Away"
+
+    ecobee_fixture["events"][0]["endDate"] = "2999-01-01"
+    assert thermostat.preset_mode == PRESET_AWAY_INDEFINITELY
+
+
+async def test_set_preset_mode(ecobee_fixture, thermostat, data) -> None:
+    """Test set preset mode."""
+    # Set a preset provided by ecobee.
+    data.reset_mock()
+    thermostat.set_preset_mode("Climate2")
```

Comment:
We shouldn't access the entity directly. We should set up a mock config entry of the integration and then make the corresponding service call via the core service registry.

https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations